src/effects/builtin/tonegen/ToneView.qml

```diff
@@ -66,15 +60,16 @@ EffectBase {
         }
 
         TextInputField {
-            width: 80
-            currentText: formatNumber(tone.frequency, 6)
+            Layout.preferredWidth: 80
+
+            currentText: tone.frequencyStart
 
-            validator: DoubleInputValidator {
-                decimal: 6
+            validator: DoubleValidator {
+                bottom: 1
```

Comment:
Where do you get this `DoubleValidator` from ? What does the `bottom` property determine ?

Reply:
This is a standard one https://doc.qt.io/qt-6/qml-qtquick-doublevalidator.html, top and bottom define a valid range. The difference with DoubleInputValidator is that the latter adds padding zeroes we don't need.